src/components/commons/SearchBar.tsx

```diff
@@ -0,0 +1,276 @@
+  const [isOpen, setIsOpen] = useState<boolean>(false)
+  const [isSearching, setIsSearching] = useState<boolean>(false)
+  const [searchResults, setSearchResults] = useState<SearchResult[] | undefined>(undefined)
+  const [refEle, setRefEle] = useState<any>()
+  const [popperEle, setPopperEle] = useState<any>()
+
+  const {
+    styles,
+    attributes
+  } = usePopper(refEle, popperEle, {
+    placement: 'bottom-start'
+  })
+
+  async function changeHandler (event): Promise<void> {
+    const query = event.target.value.trim()
+
+    if (query.length > 0) {
+      setIsSearching(true)
+      const results = await getSearchResults(api, network, query)
+      setSearchResults(results)
+      setIsSearching(false)
+    } else {
+      setSearchResults([])
+    }
+  }
+
+  const onChangeDebounceHandler = useMemo(() => debounce(changeHandler, 200), [])
+
+  const inputEle = useRef<HTMLInputElement>(null)
+  useEffect(() => {
+    if (isOpen && inputEle.current !== null) {
+      inputEle.current.focus()
+      setIsOpen(false)
+    }
+  }, [isOpen])
+
+  return (
+    <Menu as={Fragment}>
+      {({ open }) => (
+        <div className={classNames('flex w-full', { 'md:w-3/4 xl:w-1/2': !props.atHeader })}>
+          <div
+            className={`flex w-full p-2 rounded-3xl h-10 bg-white border ${isActive ? 'border-primary-200' : ''}`}
+            data-testid='SearchBar'
+            ref={setRefEle}
+          >
+            <Menu.Button as={Fragment}>
+              <div className='flex w-full'>
+                <IoSearchSharp size={22} className='text-gray-600 ml-0.5 self-center' />
+                <input
+                  placeholder='Search Block / Txn / Vault ID and more'
+                  className='ml-1.5 w-full focus:outline-none'
+                  data-testid='SearchBar.Input'
+                  onChange={onChangeDebounceHandler}
+                  onFocus={() => {
+                    setIsActive(true)
+                  }}
+                  onBlur={() => {
+                    setIsActive(false)
+                    setIsOpen(open)
+                  }}
+                  ref={inputEle}
+                />
+              </div>
+            </Menu.Button>
+          </div>
+
+          <Transition
+            enter='transition ease-in duration-150'
+            enterFrom='opacity-0 translate-y-1'
+            enterTo='opacity-100 translate-y-0'
+            leave='transition ease-in duration-150'
+            leaveFrom='opacity-100 translate-y-0'
+            leaveTo='opacity-0 translate-y-1'
+            show={isActive || open}
+          >
+            <div
+              className='z-40' ref={setPopperEle} style={{
+                ...styles.popper,
+                minWidth: refEle?.scrollWidth,
+                maxWidth: refEle?.scrollWidth
+              }} {...attributes.popper}
+            >
+              <div className='w-full mt-1.5 py-3 rounded-md shadow-lg filter drop-shadow bg-white z-10 overflow-hidden'>
```

Comment:
```suggestion
              <div className='w-full mt-1.5 rounded-md shadow-lg filter drop-shadow bg-white z-10 overflow-hidden'>
```

Should not add `py-3` since the rows are already padding, it makes it unnatural.